### PR TITLE
Update surelog

### PR DESCRIPTION
**Pull request by RRozak**

### CODE REVIEW

File: src/UhdmAst.cpp [hunk 1]
```diff
@@ -2635,36 +2654,48 @@ AstNode* visit_object(vpiHandle obj_h, UhdmShared& shared) {
...
+        while (vpiHandle index_h = vpi_scan(index_itr)) {
+            // With current Surelog, index operation inside vpiIndex
+            // can mean either the index operation on the root object
+            // or the operation nested in index, like a[b[c]].
+            // We distinguish it by checking the name of parent field
+            bool validParentName = false;
+            vpiHandle parent_h = vpi_handle(vpiParent, index_h);
+            if (parent_h) {
+                std::string indexParentName = get_object_name(parent_h, {vpiName, vpiFullName});
+                if (indexParentName != objectName) {
```
Comment (RRozak) [inline src/UhdmAst.cpp:2695]: It should be changed to `!vpi_compare_objects()` when https://github.com/chipsalliance/UHDM/issues/603 will be fixed.

**Comment by RRozak** [comment]
Errors from opentitan about assignment patterns can be reproduced by this module test: https://github.com/chipsalliance/UHDM-integration-tests/pull/638.
I will now try to find a minimal example.

**Comment by RRozak** [comment]
The errors can be reproduced by https://github.com/chipsalliance/UHDM-integration-tests/pull/639

**Comment by RRozak** [comment]
Blocked by https://github.com/chipsalliance/Surelog/issues/2388

**Comment by RRozak** [comment]
I think that parts of this PR (moving to separate functions etc) can be moved to another PR. I will try to do this in the next week if this PR will be still blocked.

**Comment by RRozak** [comment]
I created separate PR for moving selects to functions: https://github.com/antmicro/verilator/pull/663

**Comment by RRozak** [comment]
I created the issue for problems with StructArray tests: https://github.com/chipsalliance/Surelog/issues/2426

**Comment by RRozak** [comment]
All but one module tests will be fixed by https://github.com/chipsalliance/UHDM-integration-tests/pull/645
The test of `spi_device` module can be reproduced by https://github.com/chipsalliance/UHDM-integration-tests/pull/646

**Comment by RRozak** [comment]
It waits for https://github.com/chipsalliance/Surelog/issues/2428

**Comment by RRozak** [comment]
I created PR that consists only minimal changes required to bump Surelog: https://github.com/antmicro/verilator/pull/670.
I will continue the work there.
The modules tests that failed will be fixed by https://github.com/chipsalliance/UHDM-integration-tests/pull/649

**Comment by RRozak** [comment]
#607 fails on `PartSelectOfPartSelectedBitSelect` test, so the changed vpiVarSelect handling is necessary.

**Comment by RRozak** [comment]
Verilator from this branch fails on this test: https://github.com/chipsalliance/UHDM-integration-tests/pull/656, but the one from uhdm-verilator passes it.

File: src/UhdmAst.cpp (hunk 1 above)
Comment (rkapuscik) [reply src/UhdmAst.cpp:2695]: It would be useful to leave a `// TODO:` then.

**Review by rkapuscik — APPROVED** [review]

**Pull request merged**